test/integration_test.go

```diff
@@ -1801,6 +1801,63 @@ func (ts *IntegrationTestSuite) TestResetWorkflowExecution() {
 	ts.Equal(originalResult, newResult)
 }
 
+func (ts *IntegrationTestSuite) TestResetWorkflowExecutionWithUpdate() {
+	ctx := context.Background()
+	wfId := "reset-workflow-execution-with-update"
+	run, err := ts.client.ExecuteWorkflow(ctx,
+		ts.startWorkflowOptions(wfId), ts.workflows.UpdateBasicWorkflow)
+	ts.NoError(err)
+	// Send a few updates to the workflow
+	for i := 0; i < 2; i++ {
+		handler, err := ts.client.UpdateWorkflow(ctx, client.UpdateWorkflowOptions{
+			WorkflowID:   run.GetID(),
+			RunID:        run.GetRunID(),
+			UpdateName:   "update",
+			Args:         []interface{}{time.Millisecond},
+			WaitForStage: client.WorkflowUpdateStageCompleted,
+		})
+		ts.NoError(err)
+		ts.NoError(handler.Get(ctx, nil))
+	}
+	// Complete workflow
+	ts.NoError(ts.client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "finish", "finished"))
+	var updatesProcessed int
+	ts.NoError(run.Get(ctx, &updatesProcessed))
+	ts.Equal(2, updatesProcessed)
+	// Reset the workflow
+	resp, err := ts.client.ResetWorkflowExecution(ctx, &workflowservice.ResetWorkflowExecutionRequest{
+		Namespace: ts.config.Namespace,
+		WorkflowExecution: &commonpb.WorkflowExecution{
+			WorkflowId: run.GetID(),
+			RunId:      run.GetRunID(),
+		},
+		Reason:                    "integration test",
+		WorkflowTaskFinishEventId: 4,
+		ResetReapplyType:          enumspb.RESET_REAPPLY_TYPE_ALL_ELIGIBLE,
```

Comment:
@dandavison The `api` docs say the default value for `ResetReapplyType` is `RESET_REAPPLY_TYPE_ALL_ELIGIBLE`, but that does not appear to be the case as if I don't specify it manually the test fails. Is this a bug?

Reply:
I think that's because you also have specified exclude. You can't specify both. One or the other.

Reply:
I throw an error on that in Core actually